kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -4269,6 +4269,28 @@ public boolean allowGetResource(String id)
 
 	} // allowGetResource
 
+	/**
+	 * Check that we should be allowing access to the context of the reference.
+	 * This is used to prevent people downloading assignment related files when a site is unpublished.
+	 * @param id The reference being looked up.
+	 * @return true if the user is allowed.
+	 */
+	boolean allowGetResourceContext(String id)
+	{
+		// id = /attachment/db52f761-fda4-447a-a757-2c83effb99c9/Assignments/6dde06a2-5f59-43a7-846b-4757a17a28a9/folders.zip
+		if (id.startsWith(ATTACHMENTS_COLLECTION)) {
+			String[] parts = id.split("/");
+			if (parts.length > 3) {
+				String siteId = parts[2];
+				// Limit our extra strictness to just assignments.
+				if (StringUtils.isNotEmpty(siteId) && "Assignments".equals(parts[3]) && m_siteService.siteExists(siteId)) {
+					return m_siteService.allowAccessSite(siteId);
+				}
+			}
+		}
+		return true;
```

Comment:
So it returns true (i.e., the user can get the resource) if the id doesn't start with the ATTACHMENTS_COLLECTION constant?

Reply:
Yeah, we only want to add the restrictions to assignments attachements.